rfc/mediachain-rfc-1.md

```diff
@@ -0,0 +1,435 @@
+date fields.
+
+Artefacts can also have associated their data stored in IPFS, so that media
+can be directly accessed from references to their Canonicals.
+If this is the case, then the artefact object will contain a link to
+the IPFS datablob in its `data` field
+
+Finally, both types carry a set of cryptographic signatures that assert their
+validity. The signatures come from _signatories_ with known keys in the
+system; any entity that contains a public key in its data can act
+as a signatory.
+
+We can make things more concrete with the following schema:
+```
+Entity = {
+ "type" : "entity"
+ "name" : <String>
+ ["key" : <Reference>]
+ "signatures" : <Signatures>
+ <Key> : <Value> ... ; entity metadata
+ }
+
+Artefact = {
+ "type" : "artefact"
+ "name" : <String>
+ ["created" : <Date>]
+ ["description" : <String>]
+ ["data" : <Reference>]
+ "signatures" : <Signatures>
+ <Key> : <Value> ... ; artefact metadata
+ }
+
+Reference = {
+ "@link" = <Canonical>
+ }
+
+Signatures = {
+ "_1" : {
+  "entity" : <Reference>
+  "signature" : <IPRSSignature>
+ }
+ "_2" : ...
+ ...
+ }
+
+IPRSSignature = {
+ "value" : <Reference>
+ }
+
+```
+
+### Chains and Links
+
+The fundamental operation of the Mediachain is to link entities and
+artefacts with labeled relationships and metadata.  The data model
+described so far is disconnected: it contains independent entities and
+artefacts. In order to track the evolving relationships between
+entities and artefacts we need to construct chains of updates for each
+entity and artefact in the datastore.
+
+The natural way to model chains in a distributed append-only store is
+through linked lists. Thus for each entity and artefact in the
+system, we also have an associated chain which initially points to the
+canonical of a global `Nil` object:
+```
+Chain = <ChainCell> | <Nil>
+ChainCell = {
+ "type" : ...
+ "chain" : <Reference>
+ <Key> : <Value> ...
+ }
+Nil = { "type" : "nil" }
+```
+
+The entity and artefact chains differ on the type of relationships
+they store. They both support metadata update cells, which simply
+provide new metadata to the base object as arbitray key-value pairs.
+
+The entity chain also allows for linking two entities, for example
+having a pseudonymous artist publicly revealing a true name that
+points to another entity in the system.
+
+The artefact chain further stores links to entities that
+represent creation and potential Intellectual Property rights,
+references in the Internet media-space, and derivative relations
+to other artefacts.
+
+Each chain cell carries also carries one or more signatures by
+signatory entities in the system.
+
+Note that the types of cells that can appear in either type of
+chain it is not limited to the types of relationships described
+above. These are merely the minimum schema to support the
+functionality of the Mediachain.
+
+Keeping this in mind, we can model the chains with the following
+schema:
+```
+EntityChainCell =
```

Comment:
Ah but there's an `ArtefactCreationCell`, this should be consistent, in either direction -- no preference